tests/drop_packets/test_drop_counters.py

```diff
@@ -0,0 +1,411 @@
+            if not cls.mtu:
+                cls.mtu = cls.default_mtu
+            if "PortChannel" in iface:
+                duthost.command("redis-cli -n 4 hset \"PORTCHANNEL|{}\" mtu {}".format(iface, mtu))["stdout"]
+            elif "Ethernet" in iface:
+                duthost.command("redis-cli -n 4 hset \"PORT|{}\" mtu {}".format(iface, mtu))["stdout"]
+            else:
+                raise Exception("Unsupported interface parameter - {}".format(iface))
+            cls.iface = iface
+        @classmethod
+        def restore_mtu(cls):
+            if cls.iface:
+                if "PortChannel" in cls.iface:
+                    duthost.command("redis-cli -n 4 hset \"PORTCHANNEL|{}\" mtu {}".format(cls.iface, cls.mtu))["stdout"]
+                elif "Ethernet" in cls.iface:
+                    duthost.command("redis-cli -n 4 hset \"PORT|{}\" mtu {}".format(cls.iface, cls.mtu))["stdout"]
+                else:
+                    raise Exception("Trying to restore MTU on unsupported interface - {}".format(cls.iface))
+
+    yield MTUConfig
+
+    MTUConfig.restore_mtu()
+
+
+def check_if_skip():
+    if pytest.SKIP_COUNTERS_FOR_MLNX:
+       pytest.SKIP_COUNTERS_FOR_MLNX = False
+       pytest.skip("Currently not supported on Mellanox platform")
+
+
+@pytest.fixture(scope='module')
+def do_test():
+    def do_counters_test(discard_group, pkt, ptfadapter, duthost, ports_info, sniff_ports, tx_dut_ports=None, comparable_pkt=None):
+        """
+        Execute test - send packet, check that expected discard counters were incremented and packet was dropped
+        @param discard_group: Supported 'discard_group' values: 'L2', 'L3', 'ACL'
+        @param pkt: PTF composed packet, sent by test case
+        @param ptfadapter: fixture
+        @param duthost: fixture
+        @param dut_iface: DUT interface name expected to receive packets from PTF
+        @param sniff_ports: DUT ports to check that packets were not egressed from
+        """
+        check_if_skip()
+        base_verification(discard_group, pkt, ptfadapter, duthost, ports_info, tx_dut_ports)
+
+        # Verify packets were not egresed the DUT
+        exp_pkt = expected_packet_mask(pkt)
+        testutils.verify_no_packet_any(ptfadapter, exp_pkt, ports=sniff_ports)
+
+    return do_counters_test
+
+
+def test_reserved_dmac_drop(do_test, ptfadapter, duthost, setup, fanouthost, pkt_fields, ports_info):
+    """
+    @summary: Verify that packet with reserved DMAC is dropped and L2 drop counter incremented
+    @used_mac_address:
+        01:80:C2:00:00:05 - reserved for future standardization
+        01:80:C2:00:00:08 - provider Bridge group address
+    """
+    reserved_mac_addr = ["01:80:C2:00:00:05", "01:80:C2:00:00:08"]
+    for reserved_dmac in reserved_mac_addr:
+        dst_mac = reserved_dmac
+
+        if "mellanox" == duthost.facts["asic_type"]:
+            pytest.SKIP_COUNTERS_FOR_MLNX = True
```

Comment:
Why use this strange trick of setting an attribute 'SKIP_COUNTERS_FOR_MLNX' to the `pytest` module and update its value by calling `check_if_skip` in`do_test`? Why not just simply call pytest.skip to skip early for this test if asic_type is "mellanox"?

Reply:
`SKIP_COUNTERS_FOR_MLNX` parameter is mainly used for the common drop test cases which appear in drop_packets.py file. 
In this function, I changed it to regular pytest.skip() call.